htdocs/takepos/ajax/ajax.php

```diff
@@ -283,6 +289,10 @@
 		$sql .= $hookmanager->resPrint;
 	}
 
+	if ($conf->global->TAKEPOS_PRODUCT_IN_STOCK == 1 && empty(getDolGlobalInt('CASHDESK_ID_WAREHOUSE'.$_SESSION['takeposterminal']))) {
```

Comment:
Use
`if (getDolGlobalInt('TAKEPOS_PRODUCT_IN_STOCK') == 1`
for php8 compatibility